splink/linker.py

```diff
@@ -469,6 +463,27 @@ def _execute_sql_against_backend(self, sql, templated_name, physical_name):
             f"_execute_sql_against_backend not implemented for {type(self)}"
         )
 
+    def _do_execute_sql(self, final_sql, templated_name, physical_name):
+        """Actually execute the sql against the backend database.
+
+        This could return something, or not. It's up to the Linker subclass to decide.
+        """
+        raise NotImplementedError(f"_do_execute_sql not implemented for {type(self)}")
+
+    def _log_and_do_execute_sql(self, final_sql, templated_name, physical_name):
+        """Log the sql, then call _do_execute_sql(), wrapping any errors"""
+        logger.debug(execute_sql_logging_message_info(templated_name, physical_name))
+        logger.log(5, log_sql(final_sql))
+        try:
+            return self._do_execute_sql(final_sql, templated_name, physical_name)
+        except Error as e:
+            raise SplinkException(
+                f"""
+                Error executing sql for {templated_name} ({physical_name}):
```

Comment:
Possibly something such as the following is slightly clearer?

```suggestion
                Error executing the following sql for table `{templated_name}` ({physical_name}):
```

Reply:
Done